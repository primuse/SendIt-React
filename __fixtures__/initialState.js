export const props = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: false
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: false
  }
};

export const props2 = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: true
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: true
  }
};

export const props3 = {
  parcel: {
    isLoading: true
  }
};
