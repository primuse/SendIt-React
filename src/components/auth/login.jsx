import React, { Component } from "react";
import { LoginForm, SignupForm } from "../forms/loginForm.jsx";
import { Link } from "react-router-dom";
import "../../css/modules.css";
import "../../css/style.css";
import "../../css/login.css";

export const Login = () => {
  return (
    <section id="login">
      <header>
        <ul>
          <li id="head-brand">
            <Link to="/">
              Send<span className="head-span">IT</span>
            </Link>
          </li>
        </ul>
        <ul className="ml-auto mt-25">
          <li>
            <Link to="/signup">Signup</Link>
          </li>
        </ul>
      </header>
      <div className="banner">
        <div className="sendit-text mb-34">
          <h5>
            The <span className="head-span">Courier</span> service app you've
            been waiting for
          </h5>
          <p>SendIT makes reaching the world easier!</p>
        </div>
        <LoginForm />
      </div>
    </section>
  );
};

export const Signup = () => {
  return (
    <section id="login">
      <header>
        <ul>
          <li id="head-brand">
            <Link to="/">
              Send<span class="head-span">IT</span>
            </Link>
          </li>
        </ul>
        <ul class="ml-auto mt-25">
          <li>
            <Link to="/">Login</Link>
          </li>
        </ul>
      </header>
      <div class="banner">
        <div class="sendit-text">
          <h5>
            The <span class="head-span">Courier</span> service app you've been
            waiting for
          </h5>
          <p class="mb-34">
            Register to experience our world class courier services
          </p>
        </div>
        <div id="form-box">
          <div class="text-center">
            <SignupForm />
          </div>
        </div>
      </div>
    </section>
  );
};
