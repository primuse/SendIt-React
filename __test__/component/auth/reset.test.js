import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Reset } from '../../../src/components/auth/login.jsx';
import { props } from '../../../__fixtures__/initialState';

const mockStore = configureStore([thunk]);
const store = mockStore(props);


describe('Reset Page component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <Reset />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
