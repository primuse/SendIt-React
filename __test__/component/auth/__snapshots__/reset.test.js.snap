// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Page component renders correctly 1`] = `
<section
  id="login"
>
  <header>
    <ul>
      <li
        id="head-brand"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Send
          <span
            className="head-span"
          >
            IT
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="ml-auto"
    >
      <li>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </li>
    </ul>
  </header>
  <div
    className="banner bg-pd"
  >
    <div
      className="sendit-text mb-24"
    >
      <h5>
        The 
        <span
          className="head-span"
        >
          Courier
        </span>
         service app you've been waiting for
      </h5>
    </div>
    <div
      id="form-box"
    >
      <form
        className="text-center"
        id="resetForm"
        onSubmit={[Function]}
      >
        <div
          className="mb-14"
        >
          <input
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-row"
        >
          <small>
            A reset link would be sent to you
          </small>
        </div>
        <div>
          <button
            className="btn sm is-outlined"
            type="submit"
          >
            Reset Password
          </button>
        </div>
      </form>
    </div>
  </div>
</section>
`;
